llvm/lib/IR/RepoHashCalculator.cpp

```diff
@@ -337,9 +337,18 @@ void HashCalculator::hashGlobalValue(const GlobalValue *V) {
   }
 
   if (auto *GO = dyn_cast<GlobalObject>(V)) {
-    // Push GO into the dependent list if it is not a declaration.
-    if (!GO->isDeclaration())
-      getDependencies().emplace_back(GO);
+    // Don't push GO into the dependent list if it is a declaration.
+    if (GO->isDeclaration())
+      return;
```

Comment:
I'm not a big fan of having a return statement in the middle of a chunk of code. This isn't quite an early return because it's not one of the first statements in the function.

How about pulling this out into a function of its own? We'd have something like:


    bool shouldAddGODependency (const GlobalObject & GO) {
        if (GO.isDeclaration()) {
            return false;
        }
        … etc. …
        return true;
    }
    … 
        if (auto * const GO = dyn_cast<GlobalObject>(V)) {
            if (shouldAddGODependency (*GO)) {
                getDependencies().emplace_back(GO);
            }
        }

Reply:
I have updated the patch following your advice. 

I changed the `shouldAddGODependency` parameter from `reference` to `pointer`.
```
static bool shouldAddGODependency (const GlobalObject * GO) {
```
Since the following statement inside of the function.
```
const llvm::Function *const Fn = dyn_cast<const llvm::Function>(GO);
```

Reply:
> I changed the shouldAddGODependency parameter from reference to pointer.

I prefer the reference version unless you’ve updated the code to do something sensible if nullptr is passed. However, please don’t feel compelled to change it back.

Reply:
> 
> I prefer the reference version unless you’ve updated the code to do something sensible if nullptr is passed. However, please don’t feel compelled to change it back.

Good point. I have changed back to use the reference. Thanks for pointing this out :).